src/diffusion/model.py

```diff
@@ -123,6 +124,9 @@ def forward(
                     input_ids=encoder_input_ids, attention_mask=encoder_attention_mask
                 ).last_hidden_state
 
+                if self.encoder_dim != self.decoder_dim:
+                    context = self.adapter(context)
+
```

Comment:
If you are using adapter only inside if expression than init it inside same expression.